src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -924,9 +924,52 @@ private static Symbol GetNearestOtherSymbol(ConsList<TypeSymbol> list, TypeSymbo
             Debug.Assert(type.IsInterface);
 
             LookupMembersWithoutInheritance(current, type, name, arity, options, originalBinder, accessThroughType, diagnose, ref useSiteDiagnostics, basesBeingResolved);
-            if ((options & (LookupOptions.NamespaceAliasesOnly | LookupOptions.NamespacesOrTypesOnly)) == 0 && !originalBinder.InCrefButNotParameterOrReturnType)
+            if ((options & LookupOptions.NamespaceAliasesOnly) == 0 && !originalBinder.InCrefButNotParameterOrReturnType)
             {
-                LookupMembersInInterfacesWithoutInheritance(current, type.AllInterfacesWithDefinitionUseSiteDiagnostics(ref useSiteDiagnostics), name, arity, basesBeingResolved, options, originalBinder, accessThroughType, diagnose, ref useSiteDiagnostics);
+                ImmutableArray<NamedTypeSymbol> baseInterfaces;
+
+                if (basesBeingResolved?.Any() != true)
+                {
+                    baseInterfaces = type.AllInterfacesWithDefinitionUseSiteDiagnostics(ref useSiteDiagnostics);
+                }
+                else
+                {
+                    var interfacesLookedAt = PooledHashSet<NamedTypeSymbol>.GetInstance();
```

Comment:
>PooledHashSet<NamedTypeSymbol>.GetInstance(); [](start = 45, length = 45)

We're using object equality for comparison, but should we compare ignoring nullability instead?
It'll be worth testing with duplicate implementation differing by nullability. I think we don't want to produce an ambiguity in that case, right?

Reply:
>  I think we don't want to produce an ambiguity in that case, right?

This question is outside of scope of this PR. The PR has nothing to do with handling nullable and has no intent to change any behavior around nullable. The hashset is used to build the same set of interfaces that AllInterfacesWithDefinitionUseSiteDiagnostics API does. Feel free to open an issue to follow up on interaction between lookup in base interfaces and nullable. 

---
In reply to: [322410943](https://github.com/dotnet/roslyn/pull/38564#discussion_r322410943) [](ancestors = 322410943)